registry/health/health.go

```diff
@@ -0,0 +1,99 @@
+package health
+
+import (
+	"encoding/json"
+	"net/http"
+	"sync"
+	"time"
+)
+
+var (
+	mutex      sync.RWMutex
+	statusKeys = make(map[string]Status)
+)
+
+// Check set the interface for a Health Check
+type Check func() Status
+
+// StatusCode type allows us to use semantically meaningful Codes.
+type StatusCode string
+
+// Represents the possible server states based on the currently recorded
+// healthchecks.
+const (
+	StatusOK      = "StatusOK"
+	StatusWarning = "StatusWarning"
+	StatusError   = "StatusError"
+)
+
+// Status represents a named status check and it's current status.
+type Status struct {
```

Comment:
Was this structure going to have a "meta" field, as proposed by #133? Or, did we find that information should be a part of expvar?